tests/admin_inlines/admin.py

```diff
@@ -342,6 +344,103 @@ class ClassAdminStackedVertical(admin.ModelAdmin):
     inlines = [ClassStackedVertical]
 
 
+# admin for #32219 and earlier functionality
+# classes for testing combinations of verbose and verbose_plural
+# settings in models and in inlines.
+class NonVerboseChildVerboseParentInline(admin.TabularInline):
+    model = NonVerboseChild
+    verbose_name = 'Inline Non-Verbose Kid'
+
+
+class VerboseChildVerboseParentInline(admin.TabularInline):
+    model = VerboseChild
+    verbose_name = 'Inline Verbose Kid'
+
+
+class VerbosePluralChildVerboseParentInline(admin.TabularInline):
+    model = VerbosePluralChild
+    verbose_name = 'Inline Verbose Plural Kid'
+
+
+class MultiVerboseChildVerboseParentInline(admin.TabularInline):
+    model = MultiVerboseChild
+    verbose_name = 'Inline Multi-Verbose Kid'
+
+
+class NonVerboseChildVerbosePluralParentInline(admin.TabularInline):
+    model = NonVerboseChild
+    verbose_name_plural = 'Inline Non-Verbose Kids'
+
+
+class VerboseChildVerbosePluralParentInline(admin.TabularInline):
+    model = VerboseChild
+    verbose_name_plural = 'Inline Verbose Kids'
+
+
+class VerbosePluralChildVerbosePluralParentInline(admin.TabularInline):
+    model = VerbosePluralChild
+    verbose_name_plural = 'Inline Verbose Plural Kids'
+
+
+class MultiVerboseChildVerbosePluralParentInline(admin.TabularInline):
+    model = MultiVerboseChild
+    verbose_name_plural = 'Inline Multi-Verbose Kids'
+
+
+class NonVerboseChildMultiVerboseParentInline(admin.TabularInline):
+    model = NonVerboseChild
+    verbose_name = 'Inline Non-Verbose Kid'
+    verbose_name_plural = 'Inline Non-Verbose Kids'
+
+
+class VerboseChildMultiVerboseParentInline(admin.TabularInline):
+    model = VerboseChild
+    verbose_name = 'Inline Verbose Kid'
+    verbose_name_plural = 'Inline Verbose Kids'
+
+
+class VerbosePluralChildMultiVerboseParentInline(admin.TabularInline):
+    model = VerbosePluralChild
+    verbose_name = 'Inline Verbose Plural Kid'
+    verbose_name_plural = 'Inline Verbose Plural Kids'
+
+
+class MultiVerboseChildMultiVerboseParentInline(admin.TabularInline):
+    model = MultiVerboseChild
+    verbose_name = 'Inline Multi-Verbose Kid'
+    verbose_name_plural = 'Inline Multi-Verbose Kids'
```

Comment:
Can we define these classes inside a test case instead? Also all tests which work without this patch should be moved to a separate commit, as additional test coverage.

Reply:
> Can we define these classes inside a test case instead? 

I gave that a try but couldn't get it to work. It also seems inconsistent with the approach for almost all the other tests in here, so I did not feel like spending more effort on it.

> Also all tests which work without this patch should be moved to a separate commit, as additional test coverage.

Sorry, not sure what you mean by a separate commit. Are you proposing a separate Pull Request to add these tests for the existing functionality of Django 3.2.

Reply:
> I gave that a try but couldn't get it to work. It also seems inconsistent with the approach for almost all the other tests in here, so I did not feel like spending more effort on it.

OK, I can try to move them with final edits. If a class is only used in the single test, there is no need to define it in "global" scope.

> Sorry, not sure what you mean by a separate commit. Are you proposing a separate Pull Request to add these tests for the existing functionality of Django 3.2.

Yes a separate PR or commit. We don't want to mix bugfixes with extra test coverage.

Reply:
Not sure if you actually want to merge this, but a PR with the same tests that reflect the existing functionality in Django 3.2 is here: https://github.com/django/django/pull/14838